src/witan/app/forecast.clj

```diff
@@ -103,9 +105,33 @@
 
 ;;;;;;
 
+;;;;;; NOTES:
+;; - We currently don't use `exists?` properly, but we should in the context of
+;;   re-naming forecasts. Currently, duplicate owner+name combinations return
+;;   304.
+
 (defresource forecasts
+  util/json-resource
   :allowed-methods #{:get :post}
-  :available-media-types ["application/json"]
-  :handle-ok (fn [_] (s/validate
-                      [ws/Forecast]
-                      (map ->Forecast (get-forecasts)))))
+  :processable? (util/post!-processable-validation ws/NewForecast)
+  :exists? (fn [ctx]
+             (if (= :post (-> ctx :request :request-method))
+               (let [{:keys [name]} (-> ctx :request :body-params)
+                     owner (-> ctx :request :identity)]
+                 (not-empty (c/exec (find-forecast-by-name-and-owner name owner))))
+               true))
```

Comment:
if you are hitting the database in exists (usually for a get) I'd put the results of that query into the context as soon as possible so that downstream functions only need to check the context and not the database.